src/config/utils.ts

```diff
@@ -0,0 +1,12 @@
+import path from "node:path";
+import * as vscode from "vscode";
+import _ from "lodash";
+import { CONFIG_GLOB, CONFIG_GLOB_EXCLUDE } from "../constants";
+
+export const findTpnConfigFile = async (): Promise<string | undefined> => {
+    const configs = (await vscode.workspace.findFiles(CONFIG_GLOB, CONFIG_GLOB_EXCLUDE)).sort((a, b) => {
+        return a.fsPath.split(path.sep).length - b.fsPath.split(path.sep).length;
+    });
+
+    return _.isEmpty(configs) ? undefined : configs[0].fsPath;
```

Comment:
> Don't use the undefined primitive. It's a value that the JS compiler will automatically set for you when you declare variables without assignment or if you try to access properties of objects for which there is no reference. On the other hand, use null if and only if you intentionally want a variable to have "no value".

So i would suggest to use `null` here.
Or, at least, get rid of explicit "undefined":
```ts
if (_.isEmpty(configs)) {
    return;
}

return configs[0].fsPath;
```